### DISCUSSION TITLE
umi4项目本地依赖缓存疑问求解

### DISCUSSION THREAD

**Discussion by GuanJdoJ** [umijs/umi#11953] [Q&A] [answered]
使用umi4创建项目，yarn add link:/local-package方式安装本地依赖，在更改local-package代码后，umi4项目中依赖代码没有实时更改，请问是什么原因导致的呢？
1 umi项目node_modules/.cache缓存导致的，如是，能否关闭对应缓存
2 yarn add link方式就不能实时更新

**Comment by GuanJdoJ** [comment]
我又在尝试了yarn link的方式：
local-package中执行yarn link，main指向了动态更新的dist；
在umi项目中执行yarn link local-package，能够使用local-package，但是仍然无法在local-package内容更新后通过刷新umi项目启动的页面进行热更新。

**Comment by fz6m** [comment, marked as answer]
我没用过 link ，因为 link 有很多问题，包括代码不及时更新，幽灵依赖等，比较好的联动开发代码方式是 pnpm workspace 。

可以尝试把这个包名配到 [`extraBabelIncludes`](https://umijs.org/docs/api/config#extrababelincludes) 里，这样他就会被额外编译。

**Reply by GuanJdoJ** [reply]
试了`extraBabelIncludes`，很遗憾也没有生效，我再研究下pnpm workspace吧，谢谢

**Comment by jaluik** [comment]
使用`yacl` 同样遇到了这个问题，缓存的包不能自定义排除吗？
